fe/src/components/WalletButton.vue

```diff
@@ -0,0 +1,43 @@
+<template>
+  <div
+    class="relative"
+    @mouseover="isDropdownOpen = true"
+    @mouseleave="isDropdownOpen = false"
+  >
+    <button
+      v-if="props.connected"
+      class="flex gap-4 justify-center items-center btn-gradient"
+    >
+      <Icon name="solar:wallet-linear" size="25" />
+      <span>{{ props.address }}</span>
+      <Icon v-if="isDropdownOpen" name="ps:down" size="20" />
+      <Icon v-else name="ps:up" size="20" />
+    </button>
+    <button v-else class="btn-gradient">Connect Wallet</button>
+    <div
+      v-if="isDropdownOpen && props.connected"
+      class="absolute pt-2 right-0"
+      @mouseover="isDropdownOpen = true"
+      @mouseleave="isDropdownOpen = false"
+    >
+      <div
+        class="flex flex-col text-base text-dark text-left rounded-md w-44 divide-y divide-disabled divide border border-disabled"
+      >
+        <button class="p-2 text-left hover:bg-gray-200">Copy Address</button>
```

Comment:
If you have same classes that appears in multiple elements you can use **@apply** from Tailwind like this 
```suggestion
        <button class="your-btn-class">Copy Address</button>
```

<style scoped>
.your-btn-class{
  @apply p-2 text-left hover:bg-gray-200;
}
</style>